bitnami/kong/templates/deployment.yaml

```diff
@@ -0,0 +1,268 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "kong.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels: {{- include "kong.labels" . | nindent 4 }}
+    app.kubernetes.io/component: server
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels: {{- include "kong.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: server
+  {{- if .Values.updateStrategy }}
+  strategy: {{- toYaml .Values.updateStrategy | nindent 4 }}
+  {{- end }}
+  template:
+    metadata:
+      labels: {{- include "kong.labels" . | nindent 8 }}
+        app.kubernetes.io/component: server
+      annotations:
+        {{- if (include "kong.createExternalDBSecret" .) }}
+        checksum/secret: {{ include (print $.Template.BasePath "/external-database-secret.yaml") . | sha256sum }}
+        {{- end }}
+        checksum/configmap-kong: {{ include (print $.Template.BasePath "/kong-script-configmap.yaml") . | sha256sum }}
+        {{- if .Values.metrics.enabled }}
+        checksum/configmap-metrics-plugin: {{ include (print $.Template.BasePath "/metrics-script-configmap.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.podAnnotations }}
+        {{- include "kong.tplValue" (dict "value" .Values.podAnnotations "context" $) | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- if .Values.ingressController.enabled }}
+      serviceAccountName: {{ include "kong.serviceAccount" . }}
+      {{- end }}
+      {{- if .Values.podSecurityContext }}
+      securityContext: {{- include "kong.tplValue" (dict "value" .Values.podSecurityContext "context" $) | nindent 8 }}
+      {{- end }}
+      {{- include "kong.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.schedulerName }}
+      schedulerName: {{ .Values.schedulerName | quote }}
+      {{- end }}
+      {{- if .Values.affinity }}
+      affinity: {{- include "kong.tplValue" (dict "value" .Values.affinity "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.nodeSelector }}
+      nodeSelector: {{- include "kong.tplValue" (dict "value" .Values.nodeSelector "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.tolerations }}
+      tolerations: {{- include "kong.tplValue" (dict "value" .Values.tolerations "context" $) | nindent 8 }}
+      {{- end }}
+    {{- if .Values.initContainers }}
+      initContainers:
+      {{- include "kong.tplValue" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
+    {{- end }}
```

Comment:
I can see some inconsistency here.
Why everything is `kong.xxx` (e.g. `kong.resources` or `kong.livenessProbe`) and these parameters are not prefixed with `kong.`: affinity, tolerations, schedulerName, nodeSelector, initContainers, etc.

Reply:
Putting it here again (damn vscode GH plugin)

Because kong refers to the kong container. resources, livenessProbes are set at the container level. That's why you can set resources and livenessProbes for the kong ingress controller sidecar.

schedulerName, nodeSelector, initContainers, sidecars are for the deployment object itself

Reply:
I think it's confusing and it's not following the same approach we followed on other charts where several components are deployed (e.g. InfluxDB, PostgreSQL HA or Thanos). If you check these other charts, you'll find there's a different folder per component and a different section of parameters per component.

I think we should follow this format